### DISCUSSION TITLE
Distortion of mesh on adding plasticity with hardening

### DISCUSSION THREAD

**Discussion by Bala-1005** [idaholab/moose#17590] [Q&A Modules: Solid mechanics] [answered]
Hello everyone,

I am working on coupling phase field and plasticity modeling. I am using Isotropic plasticity with power hardening rule. But for some reason, the mesh gets distorted as shown in the figure. I have tried different boundary conditions but all the simulation runs show distortion and then stops converging. What could be the reason for this? The simulation runs perfectly when no plasticity is involved.

I use a SMALL strain definition with incremental strain for tensor mechanics.
![distortion](https://user-images.githubusercontent.com/75709531/114481338-5a93b880-9bd2-11eb-838a-3b97260a012a.png)
Thanks,
Bala

**Comment by WilkAndy** [comment]
That is a cool looking result, @Bala-1005 .   Could it be the usual mesh-dependence of strain-softening/hardening ?  Eg, with a coarse mesh, no quadpoints yield, but with a fine mesh a quadpoint yields, which then causes another quadpoint to yield, etc, leading to a substantially different result.

**Reply by Bala-1005** [reply]
Thank you @WilkAndy . I am not familiar with the concept of mesh-dependence of strain hardening. I might have to read up on it. In the mean time, would removing adaptivity of the mesh help in reducing the distortion?

**Reply by WilkAndy** [reply]
Give it a try!

**Reply by Bala-1005** [reply]
That does not seem to work as the solve stops converging really fast.

**Reply by WilkAndy** [reply]
Oh well - it was just an idea - i don't know about phase-field, so will leave this problem to an expert.

**Comment by hugary1995** [comment]
Can you let us know why you don't want the distortion (or why you think it's nonphysical)? Also can you post contour plots of displacements and effective inelastic strain?

**Reply by Bala-1005** [reply]
The problem is a solidification problem. The red part is solid and the blue part is the liquid. Therefore a distorted mesh does not seem to make a lot of sense in the liquid part. If that were the case, then I believe my elastic simulations also would have shown such a distortion which was not the case. I hope I make sense.

Here are the displacements, and effective strain plots.
![displacements](https://user-images.githubusercontent.com/75709531/114503436-5039e480-9bfb-11eb-9440-68cb1572e6ca.png)
![eff_pls_strain](https://user-images.githubusercontent.com/75709531/114503438-50d27b00-9bfb-11eb-95c4-8066ba41007e.png)

**Reply by WilkAndy** [reply]
I think it'd be useful to defined "distorted".  Eg, do you mean:

- asymmetric (there isn't exact 8-fold symmetry in your results), or
- should-be-circular-but-isn't, or
- should-be-square-but-isn't, or
- should not have the inwards-pointing fingers, or
- something else?

**Reply by Bala-1005** [reply]
I mean I expected the mesh to stay square-shaped but close to the interface, the mesh does not seem to be that way. In contrast, the results with only elasticity as shown below, shows very little to no change in shape.
![eta](https://user-images.githubusercontent.com/75709531/114560285-8a76a680-9c3a-11eb-8af5-10a25eea0295.png)

**Reply by WilkAndy** [reply]
Thanks @Bala-1005 .  Before quitting this discussion (due to my ignorance), i want to ask: are you sure the result isn't correct?  More specifically: are you sure your physics with hardening has a unique solution?

An example of a system with a non-unique solution is the stretching of a 1D material that is governed by a strain-softening law.  As you stretch the material, you'll get closer to the yield strength.  When you hit the yield strength, the whole material won't suddenly fail: just some "random" part of it will fail.  In real life, the "random" part is the part that, for whatever reason, was slightly weaker than the rest.  In a numerical model, the "random" part will be a quadpoint that was under slightly more stress, due to roundoff error (or other precision-related artifacts).  Then, because of the strain softening, that "random" part will keep deforming plastically while you continue to stretch the material, while the remainder of the 1D bar will behave elastically.

To me, your situation looks analogous to this.  There are a few "random" quadpoints that behave a little differently from the others, causing the "distortion".  You'll be able to determine whether your situation falls has this type of non-unique solution by doing a similar thought experiment to the previous paragraph.  Then, follow through the consequences of the "random" quadpoints failing: does that mean their neighbours then preferentially fail?  If so, then your "distortion" is mostly likely caused by non-uniqueness, in my opinion.

Just throwing this to you for consideration, not because i know anything about your physics or numerics!!

**Reply by Bala-1005** [reply]
Thank you for the analysis @WilkAndy .  It has helped me understand my problem better. I have been thinking in the same lines for my problem. The only point where I am not able to physically explain is the fact that the red phase is the solid part and the blue phase is the liquid part. Mesh "distortion" occurs only in the liquid phase. This can be seen better with some user defined anisotropy in the problem as shown below in the figure. Here there is a mesh "distortion" in the liquid phase between two solid phases. There are no stresses in the liquid phase as their young's modulus is several orders lower than the solids. Say, for example one of the quad points in the solid phase yields, it does not make physical sense for me to say that a neighboring quad point in the liquid has yielded as well. In other words, my question would be, Is it proper to use plasticity modeling for the entire domain or does it make more sense to use plasticity modeling only for the solid part?
![mesh](https://user-images.githubusercontent.com/75709531/115827535-65bcc480-a3da-11eb-8611-22fb7debbf15.PNG)
Thank you so much for the explanation.

Bala

**Reply by WilkAndy** [reply]
So, just checking - the "distortion" you're referring to is in the blue elements between the tips of the red regions.  If so, then i would look at numerical problems such as shear (or volumetric) locking, since those strangely shaped elements should be more square according to my eyes. 

Secondly, when you have a large contrast in young's modulus, you'll get a large contrast in residuals.  So the contribution to the overall residual from the "blue" region could be much smaller than from the "red" region, meaning the blue region might not have converged as well as the red region.  Put another way: it doesn't "cost" much (energy or force) to move a blue node compared with a "red" node.

I don't know anything about your model, but generally liquids are not governed by solid-mechanics plasticity (J2).  Plasticity means that, for example, pulling a material past its yield point and then releasing it again results in a permanently extended material - it does not return to its original shape.  I'm not aware of any fluids that act like that, or when you say "plastic fluid" are you meaning a non-Newtonian fluid?  I've obviously exceeded the limit of my knowledge in this question!

**Comment by ngrilli** [comment, marked as answer]
Dear @Bala-1005 

What is the difference between the mechanical properties of the solid and liquid?
I get this kind of problems when one region has a stiffness which is much smaller than the other.
Could you share the input file?

Best Regards,
Nicolò Grilli

**Reply by Bala-1005** [reply]
Hi @ngrilli ,
The young's moduli of the solid and liquid phases have a difference in the order of around ~10e9. Did you get these kinds of problems with plasticity or with elasticity?
Here is the input file.

Thanks,
Bala
[kks_anisotropic_plastic.txt](https://github.com/idaholab/moose/files/6304163/kks_anisotropic_plastic.txt)

**Comment by jiangwen84** [comment]
Let me tell you a tangential issue here. When you use a plastic model (with stateful properties), you'd better not use adaptivity. It does not guarantee the stress state is always admissible before and after refinement/coarsening, i.e. remain at yield surface. 

I agree with @ngrilli , the issue seems to be related to the large material contrast.
